src/app/vm/container/vm-detail.container.ts

```diff
@@ -88,12 +93,26 @@ export class VmDetailContainerComponent implements OnInit {
     });
   }
 
-  public changeAffinityGroup(affinityGroupId: string) {
+  public changeAffinityGroup(affinityGroupIds: string[]) {
     this.vm$.pipe(take(1)).subscribe((vm: VirtualMachine) => {
       this.store.dispatch(
         new vmActions.ChangeAffinityGroup({
           vm,
-          affinityGroupId,
+          affinityGroupIds,
+        }),
+      );
+    });
+  }
+
+  public removeAffinityGroup(affinityGroupId: string) {
+    this.vm$.pipe(take(1)).subscribe((vm: VirtualMachine) => {
+      const affinityGroupIds = vm.affinitygroup
+        .filter(g => g.id !== affinityGroupId)
```

Comment:
It makes sense to `map` first, then `filter(id => id !== affinityGroupId)`, but no big deal